File: src/main/java/com/example/demo/service/IndicatorService.java
package com.example.demo.service;


import java.util.Comparator;
import java.util.List;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;
import com.example.demo.repository.IndicatorRepository;
import com.example.demo.view.IndicatorView;


@Service
public class IndicatorService {

    private final IndicatorRepository indicatorRepository;

    @Autowired
    public IndicatorService(IndicatorRepository indicatorRepository) {
        this.indicatorRepository = indicatorRepository;
    }

    public List<IndicatorView> getAllIndicatorsWithAverageValue(String order) {
        List<IndicatorView> indicators = indicatorRepository.findAllWithAvgValue();

        if (order == null) {
            order = "";
        }

        switch (order) {
            case "asc": {
                indicators.sort(Comparator.comparingDouble(IndicatorView::averageKpi));
                break;
            }
            case "desc": {
                indicators.sort((o1, o2) -> -Double.compare(o1.averageKpi(), o2.averageKpi()));
                break;
            }
            default: {
                indicators.sort(Comparator.comparing(IndicatorView::name));
                break;
            }
        }
        return indicators;
    }
}

File: src/main/kotlin/com/example/demo/service/IndicatorService.kt
package com.example.demo.service

import com.example.demo.repository.IndicatorRepository
import com.example.demo.view.IndicatorView
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Service


@Service
class IndicatorService @Autowired constructor(
    private val indicatorRepository: IndicatorRepository) {

    fun getAllIndicatorsWithAverageValue(order: String?): List<IndicatorView> {

        val indicators: List<IndicatorView> = indicatorRepository.findAllWithAvgValue()

        when (order) {
            "asc" -> {
                indicators.sortedWith(Comparator.comparingDouble(IndicatorView::averageKpi))
            }
            "desc" -> {
                indicators.sortedWith { o1: IndicatorView, o2: IndicatorView ->
                    -o1.averageKpi.compareTo(o2.averageKpi)
                }
            }
            else -> {
                indicators.sortedWith(Comparator.comparing(IndicatorView::name))
            }
        }

        return indicators
    }
}